packages/notifier/src/publish-kit.js

```diff
@@ -304,32 +319,30 @@ const advanceDurablePublishKit = (context, value, targetStatus = 'live') => {
   if (done || valueDurability === 'mandatory') {
     canBeDurable(value) || Fail`Cannot accept non-durable value: ${value}`;
   }
-  const { tailP: currentP, tailR: resolveCurrent } =
+  const { tailP: oldTailP, tailR: resolveOldTail } =
     provideDurablePublishKitEphemeralData(state, facets);
+  assert.typeof(resolveOldTail, 'function');
 
   const publishCount = state.publishCount + 1n;
   state.publishCount = publishCount;
+
   let tailP;
   let tailR;
-
   if (done) {
     state.status = targetStatus;
-    tailP = makeQuietRejection(new Error('Cannot read past end of iteration.'));
+    tailP = makeTooFarRejection();
     tailR = undefined;
   } else {
     ({ promise: tailP, resolve: tailR } = makePromiseKit());
     void E.when(tailP, sink, sink);
   }
-  durablePublishKitEphemeralData.set(
-    getEphemeralKey(facets),
-    harden({ currentP, tailP, tailR }),
-  );
 
+  let currentP;
   if (targetStatus === 'failed') {
     state.hasValue = true;
     state.value = value;
```

Comment:
Could this not throw if the fail `reason` is not durable? I think we would get into an inconsistent state then?

Same below for a final value (hidden because not changed)

Reply:
Oh I missed that!